store/tikv/client_batch.go

```diff
@@ -365,18 +390,31 @@ func (a *batchConn) batchSendLoop(cfg config.TiKVClient) {
 		if length == 0 {
 			continue // All requests are canceled.
 		}
-		maxBatchID := atomic.AddUint64(&batchCommandsClient.idAlloc, uint64(length))
+
+		// Choose a connection by round-robbin.
+		var cli *batchCommandsClient
+		for {
+			a.index = (a.index + 1) % uint32(len(a.batchCommandsClients))
+			cli = a.batchCommandsClients[a.index]
+			// The lock protects the batchCommandsClient from been closed while it's inuse.
+			if cli.tryLockForSend() {
+				break
+			}
+		}
+
+		maxBatchID := atomic.AddUint64(&cli.idAlloc, uint64(length))
 		for i := 0; i < length; i++ {
 			requestID := uint64(i) + maxBatchID - uint64(length)
 			requestIDs = append(requestIDs, requestID)
 		}
-
 		req := &tikvpb.BatchCommandsRequest{
 			Requests:   requests,
 			RequestIds: requestIDs,
 		}
 
-		batchCommandsClient.send(req, entries)
+		cli.send(req, entries)
+		tikvTransportLayerLoad = atomic.LoadUint64(cli.tikvTransportLayerLoad)
+		cli.unlockForSend()
```

Comment:
maybe better use `defer` to protect someone introduces panic in future

Reply:
Done